### PR TITLE
WRP-1058: Fixed enact test `--watch` option is not working with the latest `jest-watch-typeahead`

### PR DESCRIPTION
**Pull request by alexandrumorariu**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [X] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
`enact test --watch` option is not working with the latest `jest-watch-typeahead`

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Removed `v8-compile-cache` dependency that did not work with ES modules and updated `jest-watch-typeahead` to its latest version.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Version 2.0.0 or later of `jest-watch-typeahead` dropped support for node 12 and 17. (https://github.com/jest-community/jest-watch-typeahead/blob/main/CHANGELOG.md)
When this PR is merged, node 14+ will be needed.

### Links
[//]: # (Related issues, references)
WRP-1058

### CODE REVIEW

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 ## unreleased
 
 * Upgraded `eslint-plugin-react` version to `^7.32.2`.
+* Removed `v8-compile-cache` dependency to fix problems related to ES modules.
+
+### test
+
+* Updated `jest-watch-typeahead` to `2.2.2`.
```
Comment (MikyungKim) [inline CHANGELOG.md:8]: Let's not log this.

Comment (MikyungKim) [inline CHANGELOG.md:8]:
```suggestion

### test

* Fixed `--watch` option is not working by removing `v8-compile-cache` module.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 * Upgraded `eslint-plugin-react` version to `^7.32.2`.
 
+### test
+
+* Fixed `--watch` option is not working by removing `v8-compile-cache` module.
```
Comment (MikyungKim) [inline CHANGELOG.md:7]: I'm sorry.
```suggestion
* Fixed `--watch` option is not working with the latest `jest-watch-typeahead` module.
```

**Pull request merged**